### PR TITLE
[CI] Enable more test configs in pre-commit

### PR DESCRIPTION
**Pull request by alexbatashev**

### CODE REVIEW

**Comment by dongkyunahn-intel** [comment]
For ESIMD_EMULATOR backend, following PR needs to be merged in intel/llvm-test-suite.

https://github.com/intel/llvm-test-suite/pull/869

**Comment by dongkyunahn-intel** [comment]
intel/llvm-test-suite repo is updated, and ESIMD_EMULATOR backend tests pass as follows.
```
Testing Time: 322.11s
  Unsupported      : 43
  Passed           : 64
  Expectedly Failed: 38

4 warning(s) in tests
[100%] Built target check-sycl-ext_intel_esimd_emulator-gpu
Scanning dependencies of target check-sycl-all
[100%] Built target check-sycl-all
```

File: devops/test_configs.json [hunk 1]
```diff
@@ -33,6 +33,28 @@
...
+      "config": "fpga_emu",
+      "name": "FPGA Emu LLVM Test Suite",
+      "runs-on": [
+        "x64"
+      ],
+      "image": "${{ inputs.intel_drivers_image }}",
+      "container_options": "-u 1001",
+      "check_sycl_all": "opencl:acc,host",
+      "cmake_args": ""
+    },
```
Comment (bader) [inline devops/test_configs.json:57]: Please, remove this from the PR. It's not used and effectively a dead code.

**Review by bader — CHANGES_REQUESTED** [review]

File: devops/test_configs.json (hunk 1 above)
Comment (alexbatashev) [inline devops/test_configs.json:57]:
```suggestion
```

File: devops/test_configs.json [hunk 2]
```diff
@@ -33,6 +33,17 @@
...
     },
+    {
+      "config": "esimd_emu",
+      "name": "ESIMD Emu LLVM Test Suite",
+      "runs-on": [
+        "x64"
+      ],
+      "image": "${{ inputs.intel_drivers_image }}",
+      "container_options": "-u 1001",
+      "check_sycl_all": "ext_intel_esimd_emulator:gpu,host",
```
Comment (bader) [inline devops/test_configs.json:44]:
```suggestion
      "check_sycl_all": "ext_intel_esimd_emulator:gpu",
```
Can we exclude host device from this run?

**Review by bader — DISMISSED** [review]

**Comment by bader** [comment]
NOTE: This PR conflicts with https://github.com/intel/llvm/pull/5758.

**Comment by dongkyunahn-intel** [comment]
ESIMD_EMULATOR build fails because of a missing change in https://github.com/intel/llvm/pull/5729. I'll share its status here.

**Comment by dongkyunahn-intel** [comment]
PR is posted for missing change - https://github.com/intel/llvm/pull/5767.

**Comment by bader** [comment]
@alexbatashev, I think we need to update this patch to align with https://github.com/intel/llvm/pull/5758.

**Comment by dongkyunahn-intel** [comment]
> ESIMD_EMULATOR build fails because of a missing change in #5729. I'll share its status here.

Fix is applied in origin/sycl.

**Comment by dongkyunahn-intel** [comment]
'Unexpected passes' from ESIMD_EMULATOR will be resolved with https://github.com/intel/llvm-test-suite/pull/910. I'll ask for merging the PR to intel/llvm-test-suite to Konst or Vladimir.

**Review by bader — CHANGES_REQUESTED** [review]
Please, merge https://github.com/intel/llvm-test-suite/pull/910 first and re-run precommit testing to make sure it passes.

**Comment by dongkyunahn-intel** [comment]
@alexbatashev , intel/llvm-test-suite is updated to eliminate 'Unexpected passes'. Please verify and merge.

**Review by bader — APPROVED** [review]

**Pull request merged**